### ISSUE TITLE
[blazor-wasm] Not launching correct OS specific browser

### ISSUE THREAD

**Issue by ar0311** [OmniSharp/omnisharp-vscode#4108]
## Issue Description ##

When specifying browser to launch using "blazorwasm" debug type, it is only respected when using the root "browser" option. It is not respected when using stanza under OS such as "windows".

## Steps to Reproduce ##

On windows, make sure Chrome is installed and set as default browser. Also make sure Edge is installed.

Edit launch.json in project using "blazorwasm" debug type, add 

"browser": "edge"

Launch and confirm Edge was launched.

Stop debugging, edit launch.json:

"windows": {
      "browser": "edge"
}

Launch and Chrome (default browser) is launched.

## Expected Behavior ##

Edge is launched.

## Actual Behavior ##

Chrome is launched.

## Environment information ##

**VSCode version**: 1.50.0
**C# Extension**: 1.23.3

<details><summary>Dotnet Information</summary>
.NET SDK (reflecting any global.json):
 Version:   5.0.100-rc.2.20480.7
 Commit:    53e0c8c7f9

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.19041
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\5.0.100-rc.2.20480.7\

Host (useful for support):
  Version: 5.0.0-rc.2.20475.5
  Commit:  c5a3f49c88

.NET SDKs installed:
  3.1.402 [C:\Program Files\dotnet\sdk]
  5.0.100-rc.2.20480.7 [C:\Program Files\dotnet\sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.App 3.1.8 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 5.0.0-rc.2.20475.17 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 3.1.8 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 5.0.0-rc.2.20475.5 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.WindowsDesktop.App 3.1.8 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 5.0.0-rc.2.20475.6 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]

To install additional .NET runtimes or SDKs:
  https://aka.ms/dotnet-download
</details>
<details><summary>Visual Studio Code Extensions</summary>

|Extension|Author|Version|
|---|---|---|
|csharp|ms-dotnettools|1.23.3|
|dotnet-test-explorer|formulahendry|0.7.4|
|git-graph|mhutchie|1.26.0|
|gitlens|eamodio|10.2.2|
|html-css-class-completion|Zignd|1.19.0|
|msbuild-project-tools|tintoy|0.3.8|
|razor-plus|austincummings|0.1.4|
|remote-ssh|ms-vscode-remote|0.55.0|
|remote-ssh-edit|ms-vscode-remote|0.55.0|
|vscode-sqlite|alexcvzz|0.9.0|;

</details>

**Comment by gregg-miskelly** [comment]
CC @captainsafia

**Comment by captainsafia** [comment]
Thanks for reporting this, @ar0311.

The `blazorwasm` type doesn't support OS-specific configuration types for the browser to launch.

@gregg-miskelly Is this something the JS debugger supports? I don't believe so but I might be incorrect here.

**Comment by gregg-miskelly** [comment]
I know that the C# extension supports it for launching a browser without debugging when using the somewhat-deprecated `launchBrowser` launch.json syntax, but I am not sure about scenarios when launching the browser under a debugger. @connor4312 do you know?

**Comment by connor4312** [comment]
js-debug supports this if you toggle the `runtimeExecutable`, but it doesn't have a `browser` field (chrome and edge are distinct launch types).

**Comment by captainsafia** [comment]
Took a look at this.

The OS-specific options are supported by the O# configuration and allow a user to override what command is used to launch the browser, for example how to use `open Google Chrome.app` in macOS.

Given that the `blazorwasm` debug configuration type uses the vscode-js-debug under the hood, I don't think it's applicable for us to expose this option.

**Comment by jtsom** [comment]
I have a default Blazor-wasm application and, on Windows, when I try to run I get an error dialog:
![image](https://user-images.githubusercontent.com/13488/97724613-81868400-1aa3-11eb-9e7e-9a2c06a6f750.png)

My default system browser is set to Brave, but I do have Chrome installed:
![image](https://user-images.githubusercontent.com/13488/97724691-9f53e900-1aa3-11eb-8b4f-94e4c1908e65.png)

The error dialog says to add an item to the launch.json specifying the "runtimeExecutable", but that setting is not recognized in the launch.json for a blazorwasm type.

If I set the "browser" field to "edge", it will correctly find and launch MS Edge browser.

**Comment by connor4312** [comment]
Where is chrome.exe located on your systems?

**Comment by jtsom** [comment]
"C:\Program Files (x86)\Google\Chrome Dev\Application\chrome.exe"

**Comment by jtsom** [comment]
If I change the name of the folder from "Chrome Dev" to "Chrome", the error message does not appear, and my Chrome does launch.

**Comment by jtsom** [comment]
Has this been released? I believe I'm on the latest nightly:
![image](https://user-images.githubusercontent.com/13488/97908875-4ab7a480-1d15-11eb-9066-29f10e312c63.png)

**Comment by connor4312** [comment]
Yes, it's been published -- works for me by setting the `runtimeExecutable: dev`

![](https://memes.peet.io/img/20-11-c98c9176-f1fb-4d4d-a177-5cba55e30d7d.png)

Are you still running into issues?

**Comment by jtsom** [comment]
Yes, still having the same error. 

However, this is my launch.json, created from the ".Net Generate Assets for Build and Debug" command:
![image](https://user-images.githubusercontent.com/13488/97909419-2b6d4700-1d16-11eb-9dd4-41cb777d647a.png)

**Comment by connor4312** [comment]
We should be able to detect that if the `PROGRAMFILES(X86)` environment variable is set to `C:\Program Files (x86)` in your extension host. You can check `process.env` in the console when running the `Toggle Developer Tools` command.

If it's not set you might be hitting https://github.com/microsoft/vscode/issues/106537

**Comment by jtsom** [comment]
Looks like it is set correctly:

![image](https://user-images.githubusercontent.com/13488/97910318-9408f380-1d17-11eb-96a3-ebb9215ebfdb.png)

**Comment by jtsom** [comment]
I'm also seeing this in the console, not sure if it's relevant :
![image](https://user-images.githubusercontent.com/13488/97910916-7720f000-1d18-11eb-80cc-725272fc3ada.png)

**Comment by connor4312** [comment]
That means the built-in extension is still installed (and probably stealing launches, which explains the issue). Make sure to do steps 1 and 2 here! https://github.com/microsoft/vscode-js-debug#nightly-extension

**Comment by jtsom** [comment]
Ok getting this now:
![image](https://user-images.githubusercontent.com/13488/97912061-4346ca00-1d1a-11eb-888a-190e1638a08d.png)
and when I try to add that item, I get:
![image](https://user-images.githubusercontent.com/13488/97912234-8739cf00-1d1a-11eb-87df-f01dc39c2182.png)

**Comment by connor4312** [comment]
Ok, that makes more sense.

@captainsafia is there a way to pass the runtimeExecutable through the blazor launch to js-debug?

**Comment by captainsafia** [comment]
@connor4312 Nope, we can add this if enough people express interest in it.

@jtsom As an alternative, you can try falling back to a two-step launch configuration by launching the application server via the .NET debugger and the front-end client by using the JS debugger.

**Comment by connor4312** [comment]
@jtsom here's another alternative that will arrive in the next js-debug nightly https://github.com/microsoft/vscode-js-debug/issues/836

**Comment by jtsom** [comment]
I will give it a try when it drops. Though will it work with the standard launch.json, created by default with the .Net Generate Assets command, given your comment:

> This can now be done via the debug.javascript.defaultRuntimeExecutable setting. Note that this happens at config resolution time, so for the moment only works with pwa- prefixed configs (e.g. if you have a launch type named node, then rename it to pwa-node)

**Comment by connor4312** [comment]
It will since blazor launches pwa-chrome internally
